src/stub/src/powerpc64le-linux.elf-main.c

```diff
@@ -248,18 +251,23 @@ do_xmap(
         Extent xo;
         size_t mlen = xo.size = phdr->p_filesz;
         char  *addr = xo.buf  =                 (char *)phdr->p_vaddr;
-        char *haddr =           phdr->p_memsz +                  addr;
         size_t frag  = (long)addr &~ PAGE_MASK;
         mlen += frag;
-        addr -= frag;
-        addr  += reloc;
-        haddr += reloc;
+        addr  += reloc - frag;
 
         if (addr != mmap(addr, mlen, prot | (xi ? PROT_WRITE : 0),
                 MAP_FIXED | MAP_PRIVATE | (xi ? MAP_ANONYMOUS : 0),
                 (xi ? -1 : fdi), phdr->p_offset - frag) ) {
             err_exit(8);
         }
+
+        char *haddr =           phdr->p_memsz + frag +           addr;
+
+        if ( xo.buf != addr  + frag ) { // pie relocatable addr ?
```

Comment:
This logic (depending on, or trying to detect -fpie relocatable) should not be needed.  -fpie is equivalent to (ET_DYN==.e_type).  All relocation ("sliding" an ET_DYN to a different address) has been handled by calling xfind_pages() at line 242, and omitting MAP_FIXED if ET_DYN.  There should be no further checking for -fpie.